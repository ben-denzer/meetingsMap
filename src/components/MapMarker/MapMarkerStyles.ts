import styled from 'styled-components';

const show = `
  height: 200px;
  width: 300px;
  z-index: 2;

  .popupWrapper {
    display: block;
  }
`;

export const MapMarkerWrapper = styled.div`
  position: relative;
  height: 20px;
  width: 20px;
  z-index: 1;

  img {
    height: 20px;
    width: auto;
    position: absolute;
    top: 0;
    left: 0;
  }

  .popupWrapper {
    display: none;
  }

  &:hover {
    ${show};
  }
  &.showPopup {
    ${show};
  }
`;
