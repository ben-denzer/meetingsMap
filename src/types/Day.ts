enum Day {
  Sunday,
  Monday,
  Tuesday,
  Wednesday,
  Thursday,
  Friday,
  Saturday,
  Any
}

export default Day;
