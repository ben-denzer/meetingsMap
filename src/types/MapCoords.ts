interface MapCoords {
  lat: number;
  lng: number;
}

export default MapCoords;
