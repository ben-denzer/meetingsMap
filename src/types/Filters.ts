import Day from './Day';

interface Filters {
  dayOfWeek: Day;
}

const mockFilters = {
  dayOfWeek: Day.Any
};

export { mockFilters };
export default Filters;
