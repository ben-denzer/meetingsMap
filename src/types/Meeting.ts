import Day from './Day';

interface Meeting {
  meetingId: number;
  locationFk: number;
  groupType: string;
  meetingName: string;
  day: Day;
  hour: number;
  min: number;
  type: string;
  open: boolean;
  meetingNotes?: string;
  mensMeeting?: boolean;
  womensMeeting?: boolean;
}

const mockMeeting = {
  meetingId: 1,
  locationFk: 12,
  groupType: 'aa',
  meetingName: 'My New Meeting',
  day: Day.Monday,
  hour: 12,
  min: 0,
  type: 'Big Book Study',
  open: true,
  meetingNotes: 'Here are some notes for the meeting'
};

export { mockMeeting };
export default Meeting;
