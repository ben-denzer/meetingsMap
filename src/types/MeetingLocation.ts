import MapCoords from './MapCoords';

interface MeetingLocation {
  locationId: number;
  clubName?: string;
  address: string;
  city: string;
  state: string;
  zip: string;
  coords: MapCoords;
  website?: string;
  phone?: string;
  locationNotes?: string;
}

const mockMeetingLocation: MeetingLocation = {
  locationId: 123,
  clubName: 'My Club',
  address: '123 Main St',
  city: 'My Town',
  state: 'NV',
  zip: '89023',
  coords: { lat: 35, lng: -115 },
  website: 'https://mywebsite.com',
  phone: '(555) 555-5555',
  locationNotes: 'here are some notes about this location'
};

export { mockMeetingLocation };
export default MeetingLocation;
