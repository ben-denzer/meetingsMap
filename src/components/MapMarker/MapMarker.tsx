import * as React from 'react';
import Meeting from '../../types/Meeting';
import MeetingLocation from '../../types/MeetingLocation';

const triangle = require('../../img/triangle.jpg');
import { MapMarkerWrapper } from './MapMarkerStyles';
import MarkerPopup from '../MarkerPopup/MarkerPopup';

interface Props {
  $hover?: boolean;
  isMobile: boolean;
  lat: number;
  lng: number;
  locationData: MeetingLocation;
  markerType: string;
  meetingsAtLocation: Meeting[];
  selectedLocation: number | null;
  selectLocation(id: number): void;
}

const markerTypes = {
  meeting: triangle
};

const MapMarker: React.SFC<Props> = (props: Props) => {
  const {
    isMobile,
    locationData,
    markerType,
    meetingsAtLocation,
    selectLocation,
    selectedLocation
  } = props;

  if (!meetingsAtLocation || !meetingsAtLocation.length) {
    return null;
  }

  const popupStatus =
    selectedLocation === locationData.locationId && isMobile ? 'showPopup' : '';

  return (
    <MapMarkerWrapper
      className={popupStatus}
      onClick={() => selectLocation(locationData.locationId)}
    >
      <img src={markerTypes[markerType]} />
      <div className="popupWrapper">
        <MarkerPopup
          isMobile={isMobile}
          locationData={locationData}
          meetingsAtLocation={meetingsAtLocation}
        />
      </div>
    </MapMarkerWrapper>
  );
};

export default MapMarker;
