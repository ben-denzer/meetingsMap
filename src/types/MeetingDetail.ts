import Meeting from './Meeting';
import MeetingLocation from './MeetingLocation';

interface MeetingDetail extends Meeting, MeetingLocation {}

export default MeetingDetail;
