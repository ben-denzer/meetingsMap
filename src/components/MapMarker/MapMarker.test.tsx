import * as React from 'react';
import { shallow } from 'enzyme';
import MapMarker from './MapMarker';
import { mockMeetingLocation } from '../../types/MeetingLocation';
import Meeting, { mockMeeting } from '../../types/Meeting';

const meetingsAtLocation: Meeting[] = [mockMeeting];
const selectLocation = jest.fn();

describe('MarkerPopup', () => {
  it('should render without crashing', () => {
    shallow(
      <MapMarker
        isMobile={false}
        lat={35}
        lng={-115}
        locationData={mockMeetingLocation}
        markerType="meeting"
        meetingsAtLocation={meetingsAtLocation}
        selectLocation={selectLocation}
        selectedLocation={null}
      />
    );
  });
  it('should match snapshot', () => {
    const wrapper = shallow(
      <MapMarker
        isMobile={false}
        lat={35}
        lng={-115}
        locationData={mockMeetingLocation}
        markerType="meeting"
        meetingsAtLocation={meetingsAtLocation}
        selectLocation={selectLocation}
        selectedLocation={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
