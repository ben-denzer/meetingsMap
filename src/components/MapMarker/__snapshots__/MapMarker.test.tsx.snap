// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerPopup should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapMarker
    lat={35}
    lng={-115}
    locationData={
      Object {
        "address": "123 Main St",
        "city": "My Town",
        "clubName": "My Club",
        "coords": Object {
          "lat": 35,
          "lng": -115,
        },
        "locationId": 123,
        "locationNotes": "here are some notes about this location",
        "phone": "(555) 555-5555",
        "state": "NV",
        "website": "https://mywebsite.com",
        "zip": "89023",
      }
    }
    markerType="meeting"
    meetingsAtLocation={
      Array [
        Object {
          "day": 1,
          "groupType": "aa",
          "hour": 12,
          "locationFk": 12,
          "meetingName": "My New Meeting",
          "meetingNotes": "Here are some notes for the meeting",
          "min": 0,
          "open": true,
          "type": "Big Book Study",
        },
      ]
    }
    selectLocation={[MockFunction]}
    selectedLocation={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          src="triangle.jpg"
        />,
        <div
          className="popupWrapper"
        >
          <MarkerPopup
            locationData={
              Object {
                "address": "123 Main St",
                "city": "My Town",
                "clubName": "My Club",
                "coords": Object {
                  "lat": 35,
                  "lng": -115,
                },
                "locationId": 123,
                "locationNotes": "here are some notes about this location",
                "phone": "(555) 555-5555",
                "state": "NV",
                "website": "https://mywebsite.com",
                "zip": "89023",
              }
            }
            meetingsAtLocation={
              Array [
                Object {
                  "day": 1,
                  "groupType": "aa",
                  "hour": 12,
                  "locationFk": 12,
                  "meetingName": "My New Meeting",
                  "meetingNotes": "Here are some notes for the meeting",
                  "min": 0,
                  "open": true,
                  "type": "Big Book Study",
                },
              ]
            }
          />
        </div>,
      ],
      "className": "",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "triangle.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MarkerPopup
            locationData={
              Object {
                "address": "123 Main St",
                "city": "My Town",
                "clubName": "My Club",
                "coords": Object {
                  "lat": 35,
                  "lng": -115,
                },
                "locationId": 123,
                "locationNotes": "here are some notes about this location",
                "phone": "(555) 555-5555",
                "state": "NV",
                "website": "https://mywebsite.com",
                "zip": "89023",
              }
            }
            meetingsAtLocation={
              Array [
                Object {
                  "day": 1,
                  "groupType": "aa",
                  "hour": 12,
                  "locationFk": 12,
                  "meetingName": "My New Meeting",
                  "meetingNotes": "Here are some notes for the meeting",
                  "min": 0,
                  "open": true,
                  "type": "Big Book Study",
                },
              ]
            }
          />,
          "className": "popupWrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "locationData": Object {
              "address": "123 Main St",
              "city": "My Town",
              "clubName": "My Club",
              "coords": Object {
                "lat": 35,
                "lng": -115,
              },
              "locationId": 123,
              "locationNotes": "here are some notes about this location",
              "phone": "(555) 555-5555",
              "state": "NV",
              "website": "https://mywebsite.com",
              "zip": "89023",
            },
            "meetingsAtLocation": Array [
              Object {
                "day": 1,
                "groupType": "aa",
                "hour": 12,
                "locationFk": 12,
                "meetingName": "My New Meeting",
                "meetingNotes": "Here are some notes for the meeting",
                "min": 0,
                "open": true,
                "type": "Big Book Study",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            src="triangle.jpg"
          />,
          <div
            className="popupWrapper"
          >
            <MarkerPopup
              locationData={
                Object {
                  "address": "123 Main St",
                  "city": "My Town",
                  "clubName": "My Club",
                  "coords": Object {
                    "lat": 35,
                    "lng": -115,
                  },
                  "locationId": 123,
                  "locationNotes": "here are some notes about this location",
                  "phone": "(555) 555-5555",
                  "state": "NV",
                  "website": "https://mywebsite.com",
                  "zip": "89023",
                }
              }
              meetingsAtLocation={
                Array [
                  Object {
                    "day": 1,
                    "groupType": "aa",
                    "hour": 12,
                    "locationFk": 12,
                    "meetingName": "My New Meeting",
                    "meetingNotes": "Here are some notes for the meeting",
                    "min": 0,
                    "open": true,
                    "type": "Big Book Study",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": "",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "triangle.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MarkerPopup
              locationData={
                Object {
                  "address": "123 Main St",
                  "city": "My Town",
                  "clubName": "My Club",
                  "coords": Object {
                    "lat": 35,
                    "lng": -115,
                  },
                  "locationId": 123,
                  "locationNotes": "here are some notes about this location",
                  "phone": "(555) 555-5555",
                  "state": "NV",
                  "website": "https://mywebsite.com",
                  "zip": "89023",
                }
              }
              meetingsAtLocation={
                Array [
                  Object {
                    "day": 1,
                    "groupType": "aa",
                    "hour": 12,
                    "locationFk": 12,
                    "meetingName": "My New Meeting",
                    "meetingNotes": "Here are some notes for the meeting",
                    "min": 0,
                    "open": true,
                    "type": "Big Book Study",
                  },
                ]
              }
            />,
            "className": "popupWrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "locationData": Object {
                "address": "123 Main St",
                "city": "My Town",
                "clubName": "My Club",
                "coords": Object {
                  "lat": 35,
                  "lng": -115,
                },
                "locationId": 123,
                "locationNotes": "here are some notes about this location",
                "phone": "(555) 555-5555",
                "state": "NV",
                "website": "https://mywebsite.com",
                "zip": "89023",
              },
              "meetingsAtLocation": Array [
                Object {
                  "day": 1,
                  "groupType": "aa",
                  "hour": 12,
                  "locationFk": 12,
                  "meetingName": "My New Meeting",
                  "meetingNotes": "Here are some notes for the meeting",
                  "min": 0,
                  "open": true,
                  "type": "Big Book Study",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
